File: fib.rb
#!/usr/bin/env ruby
#---------------------------------------------------------------------
# BigNum
# $ ruby fib.rb [NUM]
#---------------------------------------------------------------------
def subFib(
	num = 0
)
	if num < 0
		return
	end

	puts "0\t0"
	if num < 1
		return
	end

	puts "1\t1"
	if num < 2
		return
	end

	fib0 = 0
	fib1 = 1
	fib2 = 0
	cnt = 1

	$BufMax = 1000 * 4
	$Buf = [$BufMax]
	$BufIndex = 0

	while num > cnt
		cnt += 1

		fib2 = fib1 + fib0
		fib0 = fib1
		fib1 = fib2

		$Buf[$BufIndex + 0] = cnt
		$Buf[$BufIndex + 1] = "\t"
		$Buf[$BufIndex + 2] = fib1
		$Buf[$BufIndex + 3] = "\n"
		$BufIndex += 4

		if $BufIndex >= $BufMax
			print $Buf.join("")
			$BufIndex = 0
		end
	end

	print $Buf.slice(0..($BufIndex - 1)).join("")
end

def main()
	num = 0
	if ARGV.length > 0
		num = ARGV[0].to_i
	end
	subFib(num)
end

main()

File: fib.py
#!/usr/bin/python3
#---------------------------------------------------------------------
# BigNum
# $ python fib.py [NUM]
#---------------------------------------------------------------------
import sys

Args = sys.argv

def subFib(
	num = 0
):
	if num < 0:
		return

	print("0\t0\n", end="", sep="")
	if num < 1:
		return

	print("1\t1\n", end="", sep="")
	if num < 2:
		return

	fib0 = 0
	fib1 = 1
	fib2 = 0
	cnt = 1

	BUF_MAX = 4 * 1000
	buf = [""] * BUF_MAX
	bufIndex = 0

	while num > cnt:
		fib2 = fib1 + fib0
		fib0 = fib1
		fib1 = fib2
		cnt += 1

		buf[bufIndex + 0] = str(cnt)
		buf[bufIndex + 1] = "\t"
		buf[bufIndex + 2] = str(fib1)
		buf[bufIndex + 3] = "\n"
		bufIndex += 4

		if(bufIndex >= BUF_MAX):
			print("".join(buf), end="", sep="")
			bufIndex = 0

	print("".join(buf[0:bufIndex]), end="", sep="")

def main():
	num = 0
	if len(Args[1]) > 0:
		num = int(Args[1])
	subFib(num)

main()
